google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java

```diff
@@ -310,88 +313,36 @@ public void testGetBlobFailNonExistingGeneration() {
     }
   }
 
-  @Test(timeout = 5000)
-  public void testListBlobsSelectedFields() throws InterruptedException {
+  @Test
+  public void testListBlobsSelectedFields() {
+    String baseName = testName.getMethodName();
+
+    String name1 = baseName + "1";
+    String name2 = baseName + "2";
 
-    String[] blobNames = {
-      "test-list-blobs-selected-fields-blob1", "test-list-blobs-selected-fields-blob2"
-    };
     ImmutableMap<String, String> metadata = ImmutableMap.of("k", "v");
-    BlobInfo blob1 =
-        BlobInfo.newBuilder(bucket, blobNames[0])
-            .setContentType(CONTENT_TYPE)
-            .setMetadata(metadata)
-            .build();
-    BlobInfo blob2 =
-        BlobInfo.newBuilder(bucket, blobNames[1])
-            .setContentType(CONTENT_TYPE)
-            .setMetadata(metadata)
-            .build();
+    BlobInfo blob1 = BlobInfo.newBuilder(bucket, name1).setMetadata(metadata).build();
+    BlobInfo blob2 = BlobInfo.newBuilder(bucket, name2).setMetadata(metadata).build();
     Blob remoteBlob1 = storage.create(blob1);
     Blob remoteBlob2 = storage.create(blob2);
-    assertNotNull(remoteBlob1);
-    assertNotNull(remoteBlob2);
+
+    ImmutableSet<Map<String, String>> expected =
+        Stream.of(remoteBlob1, remoteBlob2)
+            .map(BlobInfo::getMetadata)
+            .collect(ImmutableSet.toImmutableSet());
+
     Page<Blob> page =
         storage.list(
             bucket.getName(),
-            Storage.BlobListOption.prefix("test-list-blobs-selected-fields-blob"),
+            Storage.BlobListOption.prefix(baseName),
             Storage.BlobListOption.fields(BlobField.METADATA));
-    // Listing blobs is eventually consistent, we loop until the list is of the expected size. The
-    // test fails if timeout is reached.
```

Comment:
Listing objects is strongly consistent[1], we don't need the manual retry loop here.

[1] https://cloud.google.com/storage/docs/consistency#strongly_consistent_operations